prototypes/emdpg_waveguide/EM_waveguide.cc

```diff
@@ -0,0 +1,2451 @@
+    std::vector<std::pair<unsigned int, unsigned int>> B_hat_FH;
+
+    // l vector for the linear form and relationships between test functions
+    std::vector<unsigned int> l_F;
+
+    // Reserve memory to avoid reallocations of each matrix
+    G_FF.reserve(dofs_per_cell_test * dofs_per_cell_test);
+    G_FI.reserve(dofs_per_cell_test * dofs_per_cell_test);
+    G_IF.reserve(dofs_per_cell_test * dofs_per_cell_test);
+    G_II.reserve(dofs_per_cell_test * dofs_per_cell_test);
+
+    B_FE.reserve(dofs_per_cell_trial_interior * dofs_per_cell_test);
+    B_IE.reserve(dofs_per_cell_trial_interior * dofs_per_cell_test);
+    B_FH.reserve(dofs_per_cell_trial_interior * dofs_per_cell_test);
+    B_IH.reserve(dofs_per_cell_trial_interior * dofs_per_cell_test);
+
+    B_hat_FE.reserve(dofs_per_cell_trial_skeleton * dofs_per_cell_test);
+    B_hat_IE.reserve(dofs_per_cell_trial_skeleton * dofs_per_cell_test);
+    B_hat_FH.reserve(dofs_per_cell_trial_skeleton * dofs_per_cell_test);
+
+    l_F.reserve(dofs_per_cell_test);
+
+    // Here we create the DPG local matrices and vector used for the assembly
+    // before condensation.
+    LAPACKFullMatrix<double> G_matrix(dofs_per_cell_test, dofs_per_cell_test);
+
+    LAPACKFullMatrix<double> B_matrix(dofs_per_cell_test,
+                                      dofs_per_cell_trial_interior);
+
+    LAPACKFullMatrix<double> B_hat_matrix(dofs_per_cell_test,
+                                          dofs_per_cell_trial_skeleton);
+
+    Vector<double> l_vector(dofs_per_cell_test);
+
+    // We create the condensation matrices.
+    LAPACKFullMatrix<double> M1_matrix(dofs_per_cell_trial_interior,
+                                       dofs_per_cell_trial_interior);
+    LAPACKFullMatrix<double> M2_matrix(dofs_per_cell_trial_interior,
+                                       dofs_per_cell_trial_skeleton);
+    LAPACKFullMatrix<double> M3_matrix(dofs_per_cell_trial_skeleton,
+                                       dofs_per_cell_trial_skeleton);
+    LAPACKFullMatrix<double> M4_matrix(dofs_per_cell_trial_interior,
+                                       dofs_per_cell_test);
+    LAPACKFullMatrix<double> M5_matrix(dofs_per_cell_trial_skeleton,
+                                       dofs_per_cell_test);
+
+    // During the calculation of matrix vector product, we require intermediary
+    // matrices and vector that we also allocate here.
+    LAPACKFullMatrix<double> tmp_matrix(dofs_per_cell_trial_skeleton,
+                                        dofs_per_cell_trial_interior);
+
+    LAPACKFullMatrix<double> tmp_matrix2(dofs_per_cell_trial_skeleton,
+                                         dofs_per_cell_trial_skeleton);
+
+    LAPACKFullMatrix<double> tmp_matrix3(dofs_per_cell_trial_skeleton,
+                                         dofs_per_cell_test);
+
+    Vector<double> tmp_vector(dofs_per_cell_trial_interior);
+    Vector<double> tmp_vector2(dofs_per_cell_test);
```

Comment:
I know these are temporary vectors, but is there a way you can give them a more descriptive name since they don't have the same size. Just something to distinguish tmp_vector and tmp_vector2